Assets/_Functionalities/FirstPersonViewer/Scripts/Movement Modus/UI/MovementModusSwitcher.cs

```diff
@@ -1,105 +1,81 @@
     {
-        [Header("Input")]
-        [SerializeField] private InputActionAsset inputMap;
-        private InputAction cycleNextAction;
-        private InputAction cyclePreviousAction;
-
-        [Header("UI")]
-        [SerializeField] private Image currentMovemodeImage;
-        [SerializeField] private Image nextMovemodeImage;
-        [SerializeField] private Image prevMovemodeImage;
-        [SerializeField] private MovementModusButton movementButtonPrefab;
-        [SerializeField] private Transform movementButtonParent;
+        private FirstPersonViewerInput input;
 
         [Header("Movement")]
-        [SerializeField] private MovementCollection movementPresets;
-        private MovementPresets currentMovement;
+        [field: SerializeField] public List<MovementPresets> MovementPresets { private set; get; }
+        public MovementPresets CurrentMovement { private set; get; }
 
         private void Awake()
         {
-            ViewerEvents.OnViewerEntered += ViewerEnterd;
-        }
-
-        private void Start()
-        {
-            cycleNextAction = inputMap.FindAction("NavigateModusNext");
-            cyclePreviousAction = inputMap.FindAction("NavigateModusPrevious");
-
-            movementPresets.presets.ForEach(preset =>
-            {
-                MovementModusButton tempButton = Instantiate(movementButtonPrefab, movementButtonParent);
-                tempButton.SetupButton(preset, this);
-            });
+            input = GetComponent<FirstPersonViewerInput>();
+            
+            ViewerEvents.OnViewerSetupComplete += ViewerEnterd;
         }
 
         private void OnDestroy()
         {
-            ViewerEvents.OnViewerEntered -= ViewerEnterd;
+            ViewerEvents.OnViewerSetupComplete -= ViewerEnterd;
         }
 
         private void ViewerEnterd()
         {
-            LoadMoveModus(0);
+            LoadMovementPreset(MovementPresets[0]);
         }
 
         private void Update()
         {
-            if (cyclePreviousAction.triggered) ChangeViewerModus(-1);
-            else if (cycleNextAction.triggered) ChangeViewerModus(1);        
+            if (input.CyclePreviousModus.triggered) ChangeViewerModus(-1);
+            else if (input.CycleNextModus.triggered) ChangeViewerModus(1);        
         }
 
         public void ChangeViewerModus(int switchDirection)
         {
             if (FirstPersonViewerInput.IsInputfieldSelected()) return;
 
-            int currentIndex = movementPresets.presets.IndexOf(currentMovement) + switchDirection;
+            int currentIndex = MovementPresets.IndexOf(CurrentMovement) + switchDirection;
 
-            if (currentIndex < 0) currentIndex = movementPresets.presets.Count - 1;
-            else if (currentIndex >= movementPresets.presets.Count) currentIndex = 0;
+            if (currentIndex < 0) currentIndex = MovementPresets.Count - 1;
+            else if (currentIndex >= MovementPresets.Count) currentIndex = 0;
```

Comment:
een snelle manier om dit soort wrapping te doen ipv met if/else is: 
`currentIndex += switchDirection%MovementPresets.Count + MovementPresets.Count`: verminder de switchdirection als deze groter is dan de count of de offset te krijgen, voeg 1x de count toe om altijd boven 0 uit te komen
`currentIndex %= MovementPresets.Count` verminder de index als deze groter is dan de count